TestSkipConsumptionForBackgroundJobs is unstable

## Flaky Test

### Which jobs are failing
```
2023-09-21T09:14:44.5603252Z --- FAIL: TestResourceManagerClientTestSuite (65.28s)
2023-09-21T09:14:44.5603629Z     --- PASS: TestResourceManagerClientTestSuite/TestAcquireTokenBucket (5.19s)
2023-09-21T09:14:44.5604018Z     --- PASS: TestResourceManagerClientTestSuite/TestBasicResourceGroupCURD (16.75s)
2023-09-21T09:14:44.5604386Z     --- PASS: TestResourceManagerClientTestSuite/TestLoadRequestUnitConfig (0.01s)
2023-09-21T09:14:44.5604767Z     --- PASS: TestResourceManagerClientTestSuite/TestRemoveStaleResourceGroup (1.13s)
2023-09-21T09:14:44.5605148Z     --- PASS: TestResourceManagerClientTestSuite/TestResourceGroupController (9.75s)
2023-09-21T09:14:44.5605597Z     --- PASS: TestResourceManagerClientTestSuite/TestResourceGroupControllerConfigChanged (2.58s)
2023-09-21T09:14:44.5606016Z     --- PASS: TestResourceManagerClientTestSuite/TestResourceManagerClientDegradedMode (2.63s)
2023-09-21T09:14:44.5606766Z     --- PASS: TestResourceManagerClientTestSuite/TestResourceManagerClientFailover (4.43s)
2023-09-21T09:14:44.5607323Z     --- PASS: TestResourceManagerClientTestSuite/TestResourcePenalty (0.05s)
2023-09-21T09:14:44.5607763Z     --- FAIL: TestResourceManagerClientTestSuite/TestSkipConsumptionForBackgroundJobs (0.16s)
2023-09-21T09:14:44.5608100Z     --- PASS: TestResourceManagerClientTestSuite/TestSwitchBurst (11.36s)
2023-09-21T09:14:44.5608456Z     --- PASS: TestResourceManagerClientTestSuite/TestWatchResourceGroup (2.40s)
```

### CI link
https://github.com/tikv/pd/actions/runs/6259625207/job/16996015136

### Reason for failure (if possible)


### Anything else

--- a/tests/integrations/mcs/resourcemanager/resource_manager_test.go
+++ b/tests/integrations/mcs/resourcemanager/resource_manager_test.go
@@ -1198,7 +1198,7 @@ func (suite *resourceManagerClientTestSuite) TestRemoveStaleResourceGroup() {
 	controller.Stop()
 }
 
-func (suite *resourceManagerClientTestSuite) TestSkipConsumptionForBackgroundJobs() {
+func (suite *resourceManagerClientTestSuite) TestCheckBackgroundJobs() {
 	re := suite.Require()
 	cli := suite.client
 
@@ -1218,7 +1218,7 @@ func (suite *resourceManagerClientTestSuite) TestSkipConsumptionForBackgroundJob
 	c, _ := controller.NewResourceGroupController(suite.ctx, 1, cli, cfg)
 	c.Start(suite.ctx)
 
-	resourceGroupName := suite.initGroups[1].Name
+	resourceGroupName := suite.initGroups[0].Name
 	re.False(c.IsBackgroundRequest(suite.ctx, resourceGroupName, "internal_default"))
 	// test fallback for nil.
 	re.False(c.IsBackgroundRequest(suite.ctx, resourceGroupName, "internal_lightning"))
@@ -1249,9 +1249,15 @@ func (suite *resourceManagerClientTestSuite) TestSkipConsumptionForBackgroundJob
 	re.NoError(err)
 	re.Contains(resp, "Success!")
 	// wait for watch event modify.
-	time.Sleep(time.Millisecond * 100)
+	testutil.Eventually(re, func() bool {
+		meta := c.GetActiveResourceGroup("default")
+		if meta != nil && meta.BackgroundSettings != nil {
+			return len(meta.BackgroundSettings.JobTypes) == 2
+		}
+		return false
+	}, testutil.WithTickInterval(50*time.Millisecond))
 
-	resourceGroupName = suite.initGroups[1].Name
+	resourceGroupName = suite.initGroups[0].Name
 	re.False(c.IsBackgroundRequest(suite.ctx, resourceGroupName, "internal_default"))
 	// test fallback for `"lightning", "ddl"`.
 	re.True(c.IsBackgroundRequest(suite.ctx, resourceGroupName, "internal_lightning"))

